app/screens/sso/sso_with_redirect_url.tsx

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import {intlShape} from 'react-intl';
+import {Linking, Text, TouchableOpacity, View} from 'react-native';
+import {SafeAreaView} from 'react-native-safe-area-context';
+import urlParse from 'url-parse';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+import {setDeepLinkURL} from '@actions/views/root';
+import FormattedText from '@components/formatted_text';
+import Loading from '@components/loading';
+import {Theme} from '@mm-redux/types/preferences';
+import Store from '@store/store';
+
+interface SSOWithRedirectURLProps {
+    customUrlScheme: string;
+    intl: typeof intlShape;
+    loginError: string;
+    loginUrl: string;
+    onCSRFToken: (token: string) => void;
+    onMMToken: (token: string) => void;
+    setLoginError: (value: string) => void;
+    theme: Theme
+}
+
+function SSOWithRedirectURL({
+    customUrlScheme = 'mattermost://',
+    intl,
+    loginError,
+    loginUrl,
+    onCSRFToken,
+    onMMToken,
+    setLoginError,
+    theme,
+}: SSOWithRedirectURLProps) {
+    const [error, setError] = React.useState<string>('');
+    const style = React.useMemo(() => getStyleSheet(theme), [theme]);
+
+    const redirectUrl = customUrlScheme + 'callback';
+
+    const init = (resetErrors?: boolean) => {
+        if (resetErrors !== false) {
+            setError('');
+            setLoginError('');
+        }
+        const parsedUrl = urlParse(loginUrl, true);
+        parsedUrl.set('query', {
+            ...parsedUrl.query,
+            redirect_to: redirectUrl,
+        });
+        const url = parsedUrl.toString();
+        Linking.canOpenURL(url).then(() => {
```

Comment:
please remove `canOpenUrl` and use `openUrl` like 

```
Linking.openURL(url).catch(() => {
```

reason being that `canOpenUrl` has some issues on iOS 14 and we will be reverting this PR https://github.com/mattermost/mattermost-mobile/pull/5007 to make sure it works correctly even with non-default browsers